src/node.rs

```diff
@@ -49,27 +118,218 @@ impl Builder {
     }
 
     pub fn port(mut self, port: u16) -> Self {
-        self.port = port;
+        self.port = Port(port);
+        self
+    }
+
+    pub fn at_directory(mut self, directory: impl Into<Directory>) -> Self {
+        self.directory = directory.into();
         self
     }
 
     /// It supposed to open node at the supplied location
-    pub async fn build(self, directory: impl Into<Directory>) -> Result<Node, ()> {
-        let _ = directory;
-        todo!()
+    pub async fn build(self) -> Result<Node, BuildError> {
+        struct SecondaryExecutorDispatch;
+
+        impl sc_executor::NativeExecutionDispatch for SecondaryExecutorDispatch {
+            type ExtendHostFunctions = ();
+
+            fn dispatch(method: &str, data: &[u8]) -> Option<Vec<u8>> {
+                cirrus_runtime::api::dispatch(method, data)
+            }
+
+            fn native_version() -> sc_executor::NativeVersion {
+                cirrus_runtime::native_version()
+            }
+        }
+
+        let Self {
+            mode: Mode::Full,
+            chain,
+            directory,
+            name,
+            port: Port(port),
+            validate,
+        } = self;
+        let mut args = vec![std::ffi::OsString::from("node-binary-place-holder")];
+
+        args.extend_from_slice(&["--state-pruning".into(), "archive".into()]);
+        match chain {
+            Chain::Gemini2a => args.extend_from_slice(&["--chain".into(), "gemini-2a".into()]),
+            Chain::Custom(chain) => match chain {},
+        };
+        match &directory {
+            Directory::Tmp => args.extend_from_slice(&["--tmp".into()]),
+            Directory::Custom(path) => args.extend_from_slice(&["--base-path".into(), path.into()]),
+            Directory::Default => (), // TODO: add default path here
```

Comment:
was this TODO supposed to be resolved in this PR or not?

Reply:
Fixed it!